### ISSUE TITLE
Play Pause Bug 

### ISSUE THREAD

**Issue by mahadydev** [florent37/Flutter-AssetsAudioPlayer#228]
Flutter version : 1.17.1
Platform : Android 
********************

Hey , thanks for making such an amazing package . I'm making an audio player app and there is a bug(i think). after I open a audio file , if I stop that track / or let the audio play until its duration , then go to another audio track , it won't register play/pause command. Its really strange issue. I've also tried with notification panel where I stopped a track and went to another audio file , it play / pause command doesn't work.

my code looks like this

```
class AudioPlayer with ChangeNotifier {
  final assetsAudioPlayer = AssetsAudioPlayer.withId('MY_UNIQUE_ID');

  //Play a song
  playSong(SongInfo songInfo) {
    assetsAudioPlayer.open(
      Audio.file(
        songInfo.filePath,
        metas: Metas(
          title: songInfo.title,
          artist: songInfo.artist,
          album: songInfo.album,
          image: MetasImage.file(songInfo.albumArtwork),
        ),
      ),
      showNotification: true,
      notificationSettings: NotificationSettings(
        customPlayPauseAction: (player) {
          player.isPlaying.value
              ? assetsAudioPlayer.pause()
              : assetsAudioPlayer.play();
        },
        stopEnabled: false,
      ),
    );

    notifyListeners();
  }

  //Play or Pause a song
  playorPauseSong() {
    assetsAudioPlayer.playOrPause();
    notifyListeners();
  }

  //Stop a song
  stopSong() {
    assetsAudioPlayer.stop();
    notifyListeners();
  }
}


```
**
and onTap i use 
```
AudioPlayer_player = Provider.of<AudioPlayer>(context);
_player.playSong(_query.songs[index]); //I get the songs path from a query class)
```
any help would be appreciated. thanks

**Comment by suyie001** [comment]
I have a same issue. After seek duration,the pause method will become not working.

**Comment by florent37** [comment]
yes it happen because of an error

I fixed it in 2.0.5+1, can you try ?

and to catch the error and add a custom behavior, you can use https://github.com/florent37/Flutter-AssetsAudioPlayer#error-handling

```
By default, on playing error, it stop the audio

BUT you can add a custom behavior

_player.onErrorDo = (handler){
  handler.player.stop();
};
Open another audio

_player.onErrorDo = (handler){
  handler.player.open(ANOTHER_AUDIO);
};
Try to open again on same position

_player.onErrorDo = (handler){
  handler.player.open(
      handler.playlist.copyWith(
        startIndex: handler.playlistIndex
      ),
      seek: handler.currentPosition
  );
};
```

**Comment by mahadydev** [comment]
Hey florent thanks a lot . Its working now. I can use play/pause now whenever I play Any audio track.
right now only error I'm getting after stopping an audio on console is  -

```
E/flutter (16827): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: NoSuchMethodError: The getter 'audio' was called on null.
E/flutter (16827): Receiver: null
E/flutter (16827): Tried calling: audio
E/flutter (16827): #0      AudioPlayer.initPlayer.<anonymous closure> (package:ayat_player_flutter_player/provider/audio_player.dart:31)
E/flutter (16827): #1      _rootRunUnary (dart:async/zone.dart:1192)
E/flutter (16827): #2      _CustomZone.runUnary (dart:async/zone.dart:1085)
E/flutter (16827): #3      _CustomZone.runUnaryGuarded (dart:async/zone.dart:987)
E/flutter (16827): #4      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339)
E/flutter (16827): #5      <optimized out> (<optimized out>:594)
E/flutter (16827): #6      <optimized out> (<optimized out>:710)
E/flutter (16827): #7      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:670)
E/flutter (16827): #8      _rootRun (dart:async/zone.dart:1180)
E/flutter (16827): #9      _CustomZone.run (dart:async/zone.dart:1077)
E/flutter (16827): #10     _CustomZone.runGuarded (dart:async/zone.dart:979)
E/flutter (16827): #11     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019)
E/flutter (16827): #12     _rootRun (dart:async/zone.dart:1184)
E/flutter (16827): #13     _CustomZone.run (dart:async/zone.dart:1077)
E/flutter (16827): #14     _CustomZone.runGuarded (dart:async/zone.dart:979)
E/flutter (16827): #15     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019)
E/flutter (16827): #16     _microtaskLoop (dart:async/schedule_microtask.dart:43)
E/flutter (16827): #17     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52)
```

**Comment by florent37** [comment]
Oh crap

On Android or ios ?

**Comment by mahadydev** [comment]
android

**Comment by florent37** [comment]
Ok I try to reproduce

**Comment by florent37** [comment]
I think i've fixed it, can you try `2.0.5+3`
